logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt

```diff
@@ -600,8 +611,40 @@ internal class UserDataSource internal constructor(
         clientDAO.isMLSCapable(userId.toDao(), clientId.value)
     }
 
+    override suspend fun getContactsAmountCached(): Either<StorageFailure, Int> = wrapStorageRequest {
+        metadataDAO.valueByKey(CONTACTS_AMOUNT_KEY)?.toInt()
+    }
+
+    override suspend fun getTeamMembersAmountCached(): Either<StorageFailure, Int> = wrapStorageRequest {
+        metadataDAO.valueByKey(TEAM_MEMBERS_AMOUNT_KEY)?.toInt()
+    }
+
+    override suspend fun setContactsAmountCached(amount: Int) =
+        metadataDAO.insertValue(CONTACTS_AMOUNT_KEY, amount.toString())
+
+    override suspend fun setTeamMembersAmountCached(amount: Int) =
+        metadataDAO.insertValue(TEAM_MEMBERS_AMOUNT_KEY, amount.toString())
+
+    override suspend fun getLastContactsDateUpdateDate(): Either<StorageFailure, Instant> = wrapStorageRequest {
+        metadataDAO.valueByKey(LAST_CONTACTS_UPDATE_KEY)?.let { Instant.parse(it) }
+    }
+
+    override suspend fun setContactsAmountCachingDate(date: Instant) =
+        metadataDAO.insertValue(LAST_CONTACTS_UPDATE_KEY, date.toString())
+
+    override suspend fun countContactsAmount(): Either<StorageFailure, Int> = wrapStorageRequest {
+        userDAO.countContactsAmount()
+    }
+
+    override suspend fun countTeamMembersAmount(): Either<StorageFailure, Int> = wrapStorageRequest {
+        userDAO.countTeamMembersAmount()
+    }
```

Comment:
can those be moved out of the UserRepository to their own, maybe a AnalyticsRepo ?